src/clp_ffi_py/ir/PyDecoder.cpp

```diff
@@ -27,20 +27,20 @@ PyDoc_STRVAR(
         "Decodes the next encoded log event from the IR stream buffered in the given decoder "
         "buffer. `decoder_buffer` must have been returned by a successfully invocation of "
         "`decode_preamble`. If `query` is provided, only the next log event matching the query "
-        "will be returned.\n"
-
+        "will be returned.\n\n"
         ":param decoder_buffer: The decoder buffer of the encoded CLP IR stream.\n"
         ":param query: A Query object that filters log events. See `Query` documents for more "
         "details.\n"
         ":param allow_incomplete_stream: If set to `True`, an incomplete CLP IR stream is not "
         "treated as an error. Instead, encountering such a stream is seen as reaching its end, and "
         "the function will return None without raising any exceptions.\n"
-        ":return: A newly created LogEvent instance representing the next decoded log event from "
+        ":raises: Appropriate exceptions with detailed information on any encountered failure.\n"
```

Comment:
Can we list all the possibilities? Otherwise we're forcing users to have a blanket try-except.

Reply:
We can't because there are too many possibilities, and some might be from the zstd library which we don't know the exact signature of the exception. Users are expected to catch Exception and extract the string message.